rust/codec-lexical/src/blocks.rs

```diff
@@ -0,0 +1,409 @@
+        RawBlock(block) => raw_block_to_lexical(block, context),
+        ThematicBreak(..) => thematic_break_to_lexical(),
+        _ => block_to_lexical_default(block),
+    }
+}
+
+fn heading_from_lexical(
+    tag: lexical::HeadingTagType,
+    children: Vec<lexical::InlineNode>,
+    context: &mut LexicalDecodeContext,
+) -> Block {
+    use lexical::HeadingTagType::*;
+    let level = match tag {
+        H1 => 1,
+        H2 => 2,
+        H3 => 3,
+        H4 => 4,
+        H5 => 5,
+        H6 => 6,
+    };
+
+    let content = inlines_from_lexical(children, context);
+
+    Block::Heading(Heading {
+        level,
+        content,
+        ..Default::default()
+    })
+}
+
+fn heading_to_lexical(heading: &Heading, context: &mut LexicalEncodeContext) -> lexical::BlockNode {
+    use lexical::HeadingTagType::*;
+    let tag = match heading.level {
+        1 => H1,
+        2 => H2,
+        3 => H3,
+        4 => H4,
+        5 => H5,
+        _ => H6,
+    };
+
+    let children = inlines_to_lexical(&heading.content, context);
+
+    match context.format {
+        Format::Koenig => lexical::BlockNode::ExtendedHeading(lexical::ExtendedHeadingNode {
+            tag,
+            children,
+            ..Default::default()
+        }),
+        _ => lexical::BlockNode::Heading(lexical::HeadingNode {
+            tag,
+            children,
+            ..Default::default()
+        }),
+    }
+}
+
+fn paragraph_from_lexical(
+    paragraph: lexical::ParagraphNode,
+    context: &mut LexicalDecodeContext,
+) -> Block {
+    let content = inlines_from_lexical(paragraph.children, context);
+
+    Block::Paragraph(Paragraph {
+        content,
+        ..Default::default()
+    })
+}
+
+fn paragraph_to_lexical(
+    paragraph: &Paragraph,
+    context: &mut LexicalEncodeContext,
+) -> lexical::BlockNode {
+    let children = inlines_to_lexical(&paragraph.content, context);
+
+    lexical::BlockNode::Paragraph(lexical::ParagraphNode {
+        children,
+        ..Default::default()
+    })
+}
+
+fn quote_from_lexical(
+    children: Vec<lexical::InlineNode>,
+    context: &mut LexicalDecodeContext,
+) -> Block {
+    let content = vec![p(inlines_from_lexical(children, context))];
+
+    Block::QuoteBlock(QuoteBlock {
+        content,
+        ..Default::default()
+    })
+}
+
+fn list_from_lexical(list: lexical::ListNode, context: &mut LexicalDecodeContext) -> Block {
+    let items = list
+        .children
+        .into_iter()
+        .map(|child| ListItem::new(vec![p(inlines_from_lexical(child.children, context))]))
+        .collect();
+
+    Block::List(List {
+        items,
+        ..Default::default()
+    })
+}
+
+fn list_to_lexical(list: &List, context: &mut LexicalEncodeContext) -> lexical::BlockNode {
+    let children = list
+        .items
+        .clone()
+        .into_iter()
+        .map(|item| list_item_to_lexical(item.content.first().unwrap(), context))
```

Comment:
Don't use `unwrap()` - it would cause a panic if content happens to be empty. Also, `clone()` here should not be necessary if  `list_item_to_lexical` was to take `&ListItem`.